lucene/core/src/java/org/apache/lucene/search/LRUQueryCache.java

```diff
@@ -832,5 +873,23 @@ public BulkScorer bulkScorer(LeafReaderContext context) throws IOException {
       return new DefaultBulkScorer(new ConstantScoreScorer(this, 0f, ScoreMode.COMPLETE_NO_SCORES, disi));
     }
 
+    // Perform a cache load asynchronously
+    // @return true if synchronous caching is needed, false otherwise
```

Comment:
I think this contract is a bit confusing. Usually `true` would indicate success, like `Lock#tryLock`. Maybe it would be less confusing if `true` meant that we successfully triggered an async cache operation?